src/scheduler/pbs_sched.cpp

```diff
@@ -835,29 +741,34 @@ wait_for_cmds()
 
 /**
  *
- * @brief sends end of cycle indication to the Server
+ * @brief sends end of cycle indication to all the servers configured
  *
- * @param[in] sconn - connection info to server
+ * @param[in] sd - socket descriptor to send end of cycle notification
  *
  * @return void
  */
 static void
-send_cycle_end(sched_svrconn *sconn)
+send_cycle_end()
 {
+	svr_conn_t *svr_conns;
+	int i;
 	static int cycle_end_marker = 0;
+	svr_conns =  static_cast<svr_conn_t *>(get_conn_svr_instances(clust_secondary_sock));
 
-	if (diswsi(sconn->secondary_sock, cycle_end_marker) != DIS_SUCCESS) {
-		log_eventf(PBSEVENT_SYSTEM | PBSEVENT_FORCE, PBS_EVENTCLASS_SCHED, LOG_ERR, __func__,
-			   "Not able to send end of cycle, errno = %d", errno);
-		goto err;
-	}
+	for (i = 0; i < get_num_servers(); i++) {
+		if (diswsi(svr_conns[i].sd, cycle_end_marker) != DIS_SUCCESS) {
```

Comment:
just checking, there might be servers which never started a cycle, is it ok to send them the end cycle message?

Reply:
Yes it is ok even if one of the servers does not start a cycle but getting cycle end notification. That is the best we can do. This has been discussed earlier with Bhroam.